processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -253,14 +254,37 @@ public static Float convertObjectToFloat(Object valObj)
     } else if (valObj instanceof Number) {
       return ((Number) valObj).floatValue();
     } else if (valObj instanceof String) {
-      try {
-        return Floats.tryParse((String) valObj);
-      }
-      catch (Exception e) {
-        throw new ParseException(e, "Unable to parse value[%s] as float", valObj);
-      }
+      return Floats.tryParse((String) valObj);
     } else {
       throw new ParseException("Unknown type[%s]", valObj.getClass());
     }
   }
+
+  /**
+   * Convert a string representing a floating point value to a long.
+   *
+   * If the floating point value is not an exact integral value (e.g. 42.0), or if the floating point value
+   * is too large to be contained within a long, this function returns null.
+   *
+   * @param floatStr string representing a floating point value
+   * @return long integral equivalent of floatStr, returns  null for non-integral floats and floats outside
+   *         of the values representable by longs
+   */
+  @Nullable
+  public static Long getIntegralFromFloatString(String floatStr)
```

Comment:
Renamed to `getLongFromDecimalString`, added a `tryParseLong` call before attempting BigDecimal conversion, removed `tryParseLong` from the `getLongFromDecimalString` call sites